src/Visualization/Python/Render1D.py

```diff
@@ -0,0 +1,277 @@
+#!/usr/bin/env python
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+# Python script to render 1D data written using SpECTRE in format v2
+
+import glob
+import h5py
+import argparse
+import sys
+import os
+import numpy as np
+import matplotlib as mpl
+mpl.use('Agg')
```

Comment:
I think this is needed on systems without a window system, e.g. supercomputers. Do you know of a good way to handle both cases?

Reply:
on minerva this script runs without error:

```py
import matplotlib.pyplot as plt
plt.plot(range(10))
plt.savefig('test.pdf')
```

did you have issues with this on wheeler?

Reply:
```
  File "./test.py", line 2, in <module>
    plt.plot(range(10))
  File "/usr/local/python/anaconda2-4.4.0/lib/python2.7/site-packages/matplotlib/pyplot.py", line 3306, in plot
    ax = gca()
  File "/usr/local/python/anaconda2-4.4.0/lib/python2.7/site-packages/matplotlib/pyplot.py", line 950, in gca
    return gcf().gca(**kwargs)
  File "/usr/local/python/anaconda2-4.4.0/lib/python2.7/site-packages/matplotlib/pyplot.py", line 586, in gcf
    return figure()
  File "/usr/local/python/anaconda2-4.4.0/lib/python2.7/site-packages/matplotlib/pyplot.py", line 535, in figure
    **kwargs)
  File "/usr/local/python/anaconda2-4.4.0/lib/python2.7/site-packages/matplotlib/backends/backend_qt5agg.py", line 44, in new_figure_manager
    return new_figure_manager_given_figure(num, thisFig)
  File "/usr/local/python/anaconda2-4.4.0/lib/python2.7/site-packages/matplotlib/backends/backend_qt5agg.py", line 51, in new_figure_manager_given_figure
    canvas = FigureCanvasQTAgg(figure)
  File "/usr/local/python/anaconda2-4.4.0/lib/python2.7/site-packages/matplotlib/backends/backend_qt5agg.py", line 242, in __init__
    super(FigureCanvasQTAgg, self).__init__(figure=figure)
  File "/usr/local/python/anaconda2-4.4.0/lib/python2.7/site-packages/matplotlib/backends/backend_qt5agg.py", line 66, in __init__
    super(FigureCanvasQTAggBase, self).__init__(figure=figure)
  File "/usr/local/python/anaconda2-4.4.0/lib/python2.7/site-packages/matplotlib/backends/backend_qt5.py", line 236, in __init__
    _create_qApp()
  File "/usr/local/python/anaconda2-4.4.0/lib/python2.7/site-packages/matplotlib/backends/backend_qt5.py", line 144, in _create_qApp
    raise RuntimeError('Invalid DISPLAY variable')
RuntimeError: Invalid DISPLAY variable
```
and this is something I see on every supercomputer I've ever used... So I'm surprised you don't see it on Minerva

Reply:
- It's up to the machine, not the script, which backend should be used. There are several ways to set the backend, the easiest is the `MPLBACKEND` env variable:
  ```
  MPLBACKEND="TkAgg" path/to/Render1D.py
  ```
  See [the docs](https://matplotlib.org/faq/usage_faq.html#what-is-a-backend) for details.
- Newer versions of Matplotlib set a reasonable backend by default, so if you're running this in Py3 on Wheeler you should be fine without configuring anything (this is what I did on Minerva).

Reply:
Unfortunately I don't think we can rely on the machine to be configured correctly... We need to be able to handle this at the script level

Reply:
What's wrong with the env variable?

Reply:
Nothing, it's just we cannot rely on the machines to set them, and I don't think it's very user friendly to have people invoke it that way...

Reply:
They only have to invoke it with the env var if they're not using Py3, which they should be :D  Py2 has reached EOL! Is Py3 unavailable on Wheeler?

The issue is that I don't see a good way to set the backend in the script while still supporting both showing a window _and_ plotting to a file on any machine. The issue is that (if I understand correctly) the backend has to be set before importing `matplotlib.pyplot`, and unless that's being done _within_ the plotting function we can't set the backend in `main` depending on whether the user provided a save-to filename.

Reply:
Python 2 will be used in the scientific community for the next several years, that's just the reality of things. We cannot require python 3 yet.

Since your the python expert, it would be nice if you figured out the solution here... We really cannot rely on anything but our own script to handle these cases... One option is just to not plot directly and always print to file.

Reply:
Here are the options. I let @wthrowe and you decide which one to follow:
1. Move the `import matplotlib.pyplot as plt` into the two `render_` functions and set a specific backend in `main`:

    ```py
    if args.output is None:
        mpl.use("TkAgg")
    else:
        mpl.use("Agg")
    ```
    Keep this in mind (from [here](https://matplotlib.org/faq/usage_faq.html#what-is-a-backend)):
    > Using `use()` will require changes in your code if users want to use a different backend. Therefore, you should avoid explicitly calling `use()` unless absolutely necessary.

2. Delete lines 13-14. Running the script with Py3 works by default. Running with Py2 may complain about the backend, in which case the user should run the script as `MPLBACKEND="Agg" Render1D.py`.

Reply:
In the first, can we not also check that the `MPILBACKEND` variable is unset and that we are using python2? Doesn't that cover the downsides you're worried about?

Reply:
The user may have a backend configured in their `~/.config/matplotlib` or similar config file, so just checking the env var is not enough. I'm not aware of a way to check if the `matplotlib.get_backend()` is the default or if it's configured by the user.